### PR TITLE
improve error message for unknown types

### LINKED ISSUE #2515: Improve error message when using an imported constructor instead of a type with the same name
In this piece of code we erroneously try to use an imported constructor instead of the homonymous type:
```gleam
import gleam/dynamic.{type Dynamic, DecodeError} as d

pub fn decode(instance_state: Dynamic) -> Result(Int, DecodeError) {
  d.int(instance_state)
}
```
The error message is the following:
```text
error: Unknown type
  ┌─ /tmp/problem/src/debug.gleam:3:55
  │
3 │ pub fn decode(instance_state: Dynamic) -> Result(Int, DecodeError) {
  │                                                       ^^^^^^^^^^^ Did you mean `BitArray`?

The type `DecodeError` is not defined or imported in this module.
```
This is technically correct but not super useful to fix the problem. The error message could be a bit more useful if the compiler can figure out that the `gleam/dynamic` module also exposes a type with the same name and the programmer has already imported the constructor.
```text
error: Unknown type
  ┌─ /tmp/problem/src/debug.gleam:3:66
  │
3 │ pub fn decode(instance_state: Dynamic) -> Result(List(Int), List(DecodeError)) {
  │                                                                  ^^^^^^^^^^^

The type `DecodeError` is not defined or imported in this module.
Hint: Maybe you wanted to import `type DecodeError` from the `gleam/dynamic` module?
```
This is just an idea, I'm not really sure about what the best wording would be here

### PR DESCRIPTION
**Pull request by giacomocavalieri**
This PR closes #2515

### CODE REVIEW

File: compiler-core/src/type_/error.rs [hunk 1]
```diff
@@ -14,6 +14,13 @@ use pretty_assertions::assert_eq;
 
 use super::FieldAccessUsage;
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct UnknownType {
+    pub location: SrcSpan,
+    pub name: EcoString,
+    pub hint: UnknownTypeHint,
+}
```
Comment (lpil) [inline compiler-core/src/type_/error.rs:22]: Is this used anywhere?

**Review by lpil — COMMENTED** [review]
Thank you! One question below

File: compiler-core/src/type_/error.rs (hunk 1 above)
Comment (giacomocavalieri) [reply compiler-core/src/type_/error.rs:22]: You're right, it's never used!

**Review by lpil — APPROVED** [review]
Thank you!!

**Pull request merged**